core-bundle/src/Resources/contao/dca/tl_module.php

```diff
@@ -634,7 +635,23 @@
 			'inputType'               => 'text',
 			'eval'                    => array('multiple'=>true, 'size'=>2, 'tl_class'=>'w50'),
 			'sql'                     => "varchar(255) NOT NULL default ''"
-		)
+		),
+		'rootPageDependentModules' => array
+		(
+			'exclude'                 => true,
+			'inputType'               => 'rootPageDependentModule',
+			'options_callback'        => array('contao.listener.data_container.root_page_dependent_module', 'onOptionsCallback'),
+			'save_callback'           => array
+			(
+				array('contao.listener.data_container.root_page_dependent_module', 'onSaveCallback')
```

Comment:
this should be done through annotation or service tag

Reply:
Done in 6c85b7b8a.